pkgs/development/python-modules/deeplabcut/default.nix

```diff
@@ -0,0 +1,136 @@
+{ lib, buildPythonPackage, fetchPypi
+, cudaSupport ? false
+, python3
+, fetchzip
+, tensorflow
+, setuptools_scm
+, curl
+, gzip
+, gnutar
+, certifi
+, chardet
+, click
+, easydict
+, h5py
+, imageio
+, ipython
+, matplotlib
+, moviepy
+, numpy
+, nvidia_x11 ? null
+, cudatoolkit ? null
+, cudnn ? null
+, opencv3
+, pandas
+, patsy
+, python
+, pyyaml
+, requests
+, ruamel_yaml
+, setuptools
+, scikitimage
+, scikitlearn
+, scipy
+, six
+, statsmodels
+, tables
+, tqdm
+, tkinter
+, wheel
+, wxPython_4_0
+}:
+
+buildPythonPackage rec {
+  pname = "deeplabcut";
+  version = "2.0.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "19gxd1nij39lbvlpsmx0ibnfcadxbk5n5a3qsyrgv68r6k6i8rk8";
+  };
+
+  resnet50 = fetchzip {
+    url = "http://download.tensorflow.org/models/resnet_v1_50_2016_08_28.tar.gz";
+    sha256 = "0n9k24ipgivvvd42h2bn26n4yjab0103ikjwhgcg668vbd5mv587";
+ };
+  resnet101 = fetchzip {
+    url = "http://download.tensorflow.org/models/resnet_v1_101_2016_08_28.tar.gz";
+    sha256 = "0h7b963px3xma936zysj54c8l930h3jyss7jv0gzbxalvxzq459x";
+ };
+
+  nativeBuildInputs = [ setuptools_scm curl gzip gnutar];
+  /* buildInputs = with pkgs; [ tk tcl]; */
+  propagatedBuildInputs = [
+    certifi
+    chardet
+    click
+    easydict
+    h5py
+    imageio
+    ipython
+    /* ipython-genutils */
+    matplotlib
+    moviepy
+    numpy
+    opencv3
+    pandas
+    patsy
+    /* python-dateutil */
+    pyyaml
+    requests
+    ruamel_yaml
+    setuptools
+    scikitimage
+    scikitlearn
+    scipy
+    six
+    statsmodels
+    tables
+    tensorflow
+    tqdm
+    tkinter
+    wheel
+    wxPython_4_0
+  ];
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "scipy~=1.1.0" "scipy" \
+      --replace "imageio==2.3.0" "imageio" \
+      --replace "six==1.11.0" "six" \
+      --replace "imageio==2.3.0" "imageio" \
+      --replace "ipython~=6.0.0" "ipython" \
+      --replace "moviepy~=0.2.3.5" "moviepy" \
+      --replace "numpy~=1.14.5" "numpy" \
+      --replace "pandas==0.21.0" "pandas" \
+      --replace "pyyaml>=4.2b1" "pyyaml" \
+      --replace "scipy~=1.1.0" "scipy" \
+      --replace "six==1.11.0" "six" \
+      --replace "statsmodels==0.9.0" "statsmodels" \
+      --replace "tables==3.4.3" "tables" \
+      --replace "wheel==0.31.1" "wheel" \
+      --replace "'opencv-python~=3.4'," "" \
+      --replace "ipython-genutils==0.2.0" "ipython-genutils" \
+      --replace "'intel-openmp'," "" \
+      --replace "python-dateutil==2.7.3" "python-dateutil" \
+      --replace "scikit-image~=0.14.0" "scikit-image" \
+      --replace "scikit-learn~=0.19.2" "scikit-learn" \
+      --replace "ruamel.yaml==0.15" "ruamel.yaml"
+  '';
+  /* checkInputs = with python3Packages; [ */
+      /* py */
+  /* ]; */
+  doCheck = false;
```

Comment:
probably want to do some tests, seeing as we disregard most of the versioning

Reply:
Looks like tests uses tk backend for matplotlib, which requires GUI:
```
ImportError: Failed to import test module: deeplabcut
Traceback (most recent call last):
  File "/nix/store/zdh16dcvjw99ybam59zd2ijb6bx138j0-python3-3.7.5/lib/python3.7/unittest/loader.py", line 470, in _find_test_path
    package = self._get_module_from_name(name)
  File "/nix/store/zdh16dcvjw99ybam59zd2ijb6bx138j0-python3-3.7.5/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/build/deeplabcut-2.1.3/deeplabcut/__init__.py", line 29, in <module>
    from deeplabcut import generate_training_dataset
  File "/build/deeplabcut-2.1.3/deeplabcut/generate_training_dataset/__init__.py", line 19, in <module>
    from deeplabcut.generate_training_dataset.labeling_toolbox import *
  File "/build/deeplabcut-2.1.3/deeplabcut/generate_training_dataset/labeling_toolbox.py", line 27, in <module>
    from deeplabcut.utils import auxiliaryfunctions
  File "/build/deeplabcut-2.1.3/deeplabcut/utils/__init__.py", line 1, in <module>
    from deeplabcut.utils.make_labeled_video import *
  File "/build/deeplabcut-2.1.3/deeplabcut/utils/make_labeled_video.py", line 34, in <module>
    mpl.use('TkAgg')
  File "/nix/store/6nd5i3czj5nm74fz74x2rzzpi9sjn2is-python3.7-matplotlib-3.1.1/lib/python3.7/site-packages/matplotlib/cbook/deprecation.py", line 307, in wrapper
    return func(*args, **kwargs)
  File "/nix/store/6nd5i3czj5nm74fz74x2rzzpi9sjn2is-python3.7-matplotlib-3.1.1/lib/python3.7/site-packages/matplotlib/__init__.py", line 1309, in use
    switch_backend(name)
  File "/nix/store/6nd5i3czj5nm74fz74x2rzzpi9sjn2is-python3.7-matplotlib-3.1.1/lib/python3.7/site-packages/matplotlib/pyplot.py", line 236, in switch_backend
    newbackend, required_framework, current_framework))
ImportError: Cannot load backend 'TkAgg' which requires the 'tk' interactive framework, as 'headless' is currently running
```

Reply:
This is on import... I don't think the application will work without a version of matploblib+tkAgg

Reply:
Should be possible to make it work with `xvfb_run`? If not, please, add a comment explaining the situation.

Reply:
added tests